patches/expo-secure-store/expo-secure-store+14.2.4+001+additional-config-options.patch

```diff
@@ -0,0 +1,1114 @@
++      let result = try set(value: value, with: key, options: options)
++
++      if !result {
++        return AuthType.none.rawValue
++      }
++
++      return getAuthType().rawValue
+     }
+ 
+     AsyncFunction("deleteValueWithKeyAsync") { (key: String, options: SecureStoreOptions) in
+-      let noAuthSearchDictionary = query(with: key, options: options, requireAuthentication: false)
+-      let authSearchDictionary = query(with: key, options: options, requireAuthentication: true)
+-      let legacySearchDictionary = query(with: key, options: options)
++      let noAuthSearchDictionary = try query(with: key, options: options, requireAuthentication: false)
++      let authSearchDictionary = try query(with: key, options: options, requireAuthentication: true)
++      let legacySearchDictionary = try query(with: key, options: options)
+ 
+       SecItemDelete(legacySearchDictionary as CFDictionary)
+       SecItemDelete(authSearchDictionary as CFDictionary)
+       SecItemDelete(noAuthSearchDictionary as CFDictionary)
+     }
+ 
+-    Function("canUseBiometricAuthentication") {() -> Bool in
+-      #if os(tvOS)
+-      return false
+-      #else
+-      let context = LAContext()
+-      var error: NSError?
+-      let isBiometricsSupported: Bool = context.canEvaluatePolicy(LAPolicy.deviceOwnerAuthenticationWithBiometrics, error: &error)
++    Function("canUseBiometricAuthentication") { () -> Bool in
++      return areBiometricsEnabled()
++    }
++  }
++
++  private func askForAuth(prompt: String) async throws {
++    let context = LAContext()
++    context.localizedReason = prompt
++    var error: NSError?
++    let isAuthenticationSupported: Bool = context.canEvaluatePolicy(
++      LAPolicy.deviceOwnerAuthentication,
++      error: &error
++    )
++    if error != nil {
++      throw error ?? RuntimeError("No authentication method available")
++    }
++    if !isAuthenticationSupported {
++      throw RuntimeError("No authentication method available")
++    }
++    let result = try await context.evaluatePolicy(
++      LAPolicy.deviceOwnerAuthentication,
++      localizedReason: context.localizedReason
++    )
+ 
+-      if error != nil {
+-        return false
++    if !result {
++      throw RuntimeError("Unable to authenticate")
++    }
++  }
++
++  private func getAuthType() -> AuthType {
++    if !areBiometricsEnabled() {
++      return AuthType.credentials
++    }
++
++    let biometryType = LAContext().biometryType
++
++    if biometryType == LABiometryType.none {
++      return AuthType.credentials
++    }
++
++    if biometryType == LABiometryType.faceID {
++      return AuthType.faceID
++    }
++
++    if biometryType == LABiometryType.touchID {
++      return AuthType.touchID
++    }
++
++    if #available(iOS 17.0, *) {
++      if biometryType == LABiometryType.opticID {
++        return AuthType.opticID
+       }
+-      return isBiometricsSupported
+-      #endif
+     }
```

Comment:
A bit more concise
```suggestion
private func getAuthType() -> AuthType {
  let biometryType = LAContext().biometryType
  switch biometryType {
  case .faceID: return .faceID
  case .touchID: return .touchID
  case .opticID: return .opticID // available since iOS 17
  case .none: fallthrough // this one continues to the next line
  @unknown default: return .credentials
  }
}
```